calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -319,23 +333,118 @@ def _apply_tags(self, profile_name, pod):
+        Returns None if no key-value exists
+        """
+        try:
+            val = pod['metadata'][key]
+            print("%s of pod %s:\n%s" % (key, pod, val))
+            return val
+        except KeyError:
+            print('No %s found in pod %s' % (key, pod))
+            return None
+
+    def _label_to_tag(self, label_key, label_value, namespace):
+        """
+        Labels are key-value pairs, tags are single strings. This function handles that translation
+        1) concatenate key and value with '='
+        2) prepend a pod's namespace followed by '/' if available
+        3) replace special characters with urllib-style escape sequence
+        :param label_key: key to label
+        :param label_value: value to given key for a label
+        :param namespace: Namespace string, input None if not available
+        :param types: (self, string, string, string)
+        :return single string tag
+        :rtype string
+        """
+        escape_seq = '_'
+
+        tag = '%s=%s' % (label_key, label_value)
+        tag = (namespace + '/' + tag) if namespace else tag
+        tag = quote(tag, safe='')
+        tag = tag.replace('%', escape_seq)
+
+        return tag
+
+    def _translate_rule(self, kube_rule, namespace):
+        """
+        Given a rule dict from Kubernetes Annotations, 
+        output a rule that is calicoctl profile compliant
+        :param rule: dict of keys relating to policy rules
+        :param namespace: string indicating namespace for pod, None if not available
+        :return: translated dict of keys relating to policy rules for Calico profile
+        :rtype: dict
+        """
+        # kube syntax : calico syntax
+        translation_dictionary = {
+            'srcPorts': 'src_ports',
+            'dstPorts': 'dst_ports',
+            'srcNet': 'src_net',
+            'dstNet': 'dst_net',
+            'action': 'action',
+            'protocol': 'protocol',
+            'icmpType': 'icmp_type'
+        }
+
+        calico_rule = dict()
+
+        # Kubernetes and Calico use different syntax, use a dictionary to translate
+        for kube_key, calico_key in translation_dictionary.iteritems():
+            if kube_key in kube_rule.keys():
+                calico_rule[calico_key] = kube_rule.pop(kube_key)
+
+        # Label selectors need to translate to tags
+        if 'labels' in kube_rule.keys():
+            # Iterate through 'labels' dict
+            for k, v in kube_rule['labels'].iteritems():
+                # For now, we should only see have src_tag key
+                if 'src_tag' not in calico_rule.keys():
+                    tag = self._label_to_tag(k, v, namespace)
+                    calico_rule['src_tag'] = tag
+                else:
+                    print "More than one label specified for policy rule. Multi-tag selection not supported in Calico"
+                    sys.exit(1)
+
+            kube_rule.pop('labels')
+
+        if kube_rule :
```

Comment:
Stick a comment here explaining that since we pop the entries from kube_rule, any remaining entries must be un-parseable.